sdk/src/types/block/slot/commitment.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use packable::{packer::SlicePacker, Packable, PackableExt};
+
+use crate::types::block::slot::{RootsId, SlotCommitmentId, SlotIndex};
+
+/// Contains a summary of a slot.
+/// It is linked to the commitment of the previous slot, which forms a commitment chain.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, derive_more::From, Packable)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct SlotCommitment {
+    /// The slot index of this commitment.
+    /// It is calculated based on genesis timestamp and the duration of a slot.
+    index: SlotIndex,
+    /// The commitment ID of the previous slot.
+    previous_slot_commitment_id: SlotCommitmentId,
+    /// A BLAKE2b-256 hash of concatenating multiple sparse merkle tree roots of a slot.
+    roots_id: RootsId,
+    /// The sum of previous slot commitment cumulative weight and weight of issuers of accepted blocks within this
+    /// slot. It is just an indication of "committed into" this slot, and can not strictly be used for evaluating
+    /// the switching of a chain.
+    cumulative_weight: u64,
```

Comment:
what about getters to those fields? Not needed?

Reply:
Added